src/test/java/org/jenkinsci/plugins/workflow/steps/scm/SCMStepTest.java

```diff
@@ -186,16 +197,26 @@ public InvalidChangelogSCM() { }
         @Override public void checkout(Run<?,?> build, Launcher launcher, FilePath workspace, TaskListener listener, File changelogFile, SCMRevisionState baseline) throws IOException, InterruptedException {
             // Unlike the superclass, which adds an XML header to the file, we just ignore the file.
         }
-        @Override public ChangeLogParser createChangeLogParser() {
-            return new ChangeLogParser() {
-                public ChangeLogSet<? extends ChangeLogSet.Entry> parse(Run build, RepositoryBrowser<?> browser, File changelogFile) throws IOException, SAXException {
-                    Jenkins.XSTREAM2.fromXML(changelogFile);
-                    throw new AssertionError("The previous line should always fail because the file is empty");
-                }
-            };
-        }
+        ChangeLogParser changeLogParser = createChangeLogParser();
         @TestExtension("scmParsesUnmodifiedChangelogFile")
         public static class DescriptorImpl extends NullSCM.DescriptorImpl { }
     }
 
+    public static class FakeChangeLogSCM extends NullSCM {
+        @DataBoundConstructor
+        public FakeChangeLogSCM() { }
+        ChangeLogParser changeLogParser = createChangeLogParser();
+        @TestExtension("scmParsesFakeChangeLogSCMChangelogFile")
+        public static class DescriptorImpl extends NullSCM.DescriptorImpl { }
+    }
+
+    public ChangeLogParser createChangeLogParser() {
+        return new ChangeLogParser() {
+            @Override public ChangeLogSet<? extends ChangeLogSet.Entry> parse(Run build, RepositoryBrowser<?> browser, File changelogFile) throws IOException, SAXException {
+                Jenkins.XSTREAM2.fromXML(changelogFile);
+                throw new AssertionError("The previous line should always fail because the file is empty");
+            }
+        };
+    }
+
```

Comment:
More confusing is that pretty much all of this stuff, as far as IntelliJ is concerned, is unused.

<img width="906" alt="image" src="https://user-images.githubusercontent.com/21689198/78050575-6f483a00-734a-11ea-842c-85a8a277d5f6.png">

I feel like it must have something to do with the `@TestExtension` annotation, but it's totally opaque to me how that actually works.

Reply:
It is unused because there's no code IntelliJ can detect that calls these - they are dynamically detected and called.  The same kind of thing happens in `github-api` for all the `GHObject` data fields that are initialized by Jackson using reflection.